# Data loading and inspection

In [27]:
%run "utilspro.py"
# Unzipping the provided dataset
with zipfile.ZipFile("data/amp-parkinsons-disease-progression-prediction_2.zip", 'r') as zip_ref:
    zip_ref.extractall("data/amp-parkinsons-disease-progression-prediction_2")

# Load each dataset into a dataframe
train_peptides = pd.read_csv("data/amp-parkinsons-disease-progression-prediction_2/train_peptides.csv")
train_proteins = pd.read_csv("data/amp-parkinsons-disease-progression-prediction_2/train_proteins.csv")
train_clinical_data = pd.read_csv("data/amp-parkinsons-disease-progression-prediction_2/train_clinical_data.csv")
supplemental_clinical_data = pd.read_csv("data/amp-parkinsons-disease-progression-prediction_2/supplemental_clinical_data.csv")


# Inspecting the first few rows of each dataset
dfs = {
    "Train Peptides": train_peptides,
    "Train Proteins": train_proteins,
    "Train Clinical Data": train_clinical_data,
    "Supplemental Clinical Data": supplemental_clinical_data
}

# Displaying the first few rows of each dataframe and setting truncation to False
pd.set_option('display.max_colwidth', None)
for name, df in dfs.items():
    print(name)
    print("/n")
    print(df.head())

Train Peptides
/n
  visit_id  visit_month  patient_id UniProt   
0     55_0            0          55  O00391  \
1     55_0            0          55  O00533   
2     55_0            0          55  O00533   
3     55_0            0          55  O00533   
4     55_0            0          55  O00533   

                                  Peptide  PeptideAbundance  
0                           NEQEQPLGQWHLS           11254.3  
1                             GNPEPTFSWTK          102060.0  
2                         IEIPSSVQQVPTIIK          174185.0  
3  KPQSAVYSTGSNGILLC(UniMod_4)EAEGEPQPTIK           27278.9  
4                            SMEQNGPGLEYR           30838.7  
Train Proteins
/n
  visit_id  visit_month  patient_id UniProt       NPX
0     55_0            0          55  O00391   11254.3
1     55_0            0          55  O00533  732430.0
2     55_0            0          55  O00584   39585.8
3     55_0            0          55  O14498   41526.9
4     55_0            0          55  O1

# Data cleaning


In [28]:
# Check for missing values in each dataset
missing_data_info = {}
for name, dataset in dfs.items():
    missing_data_info[name] = dataset.isnull().sum()

missing_data_info

{'Train Peptides': visit_id            0
 visit_month         0
 patient_id          0
 UniProt             0
 Peptide             0
 PeptideAbundance    0
 dtype: int64,
 'Train Proteins': visit_id       0
 visit_month    0
 patient_id     0
 UniProt        0
 NPX            0
 dtype: int64,
 'Train Clinical Data': visit_id                                  0
 patient_id                                0
 visit_month                               0
 updrs_1                                   1
 updrs_2                                   2
 updrs_3                                  25
 updrs_4                                1038
 upd23b_clinical_state_on_medication    1327
 dtype: int64,
 'Supplemental Clinical Data': visit_id                                  0
 patient_id                                0
 visit_month                               0
 updrs_1                                 213
 updrs_2                                 214
 updrs_3                                   5
 updrs_4

# Summary of missing values

1. Train Peptides:No missing values.
2. Train Proteins:No missing values.
3. Train Clinical Data:
    * updrs_1: 1 missing value
    * updrs_2: 2 missing values
    * updrs_3: 25 missing values
    * updrs_4: 1038 missing values
    * upd23b_clinical_state_on_medication: 1327 missing values

# Checking for duplicated entries in each dataset


In [31]:
# Checking for duplicate entries in each dataset
duplicate_data_info = {}
for name, dataset in dfs.items():
    duplicate_data_info[name] = dataset.duplicated().sum()

duplicate_data_info

{'Train Peptides': 0,
 'Train Proteins': 0,
 'Train Clinical Data': 0,
 'Supplemental Clinical Data': 0}

None of the datasets have duplicate entries

# Checking for datatypes

In [33]:
# Check data types for each column in every dataset
data_types_info = {}
for name, dataset in dfs.items():
    data_types_info[name] = dataset.dtypes
data_types_info

{'Train Peptides': visit_id             object
 visit_month           int64
 patient_id            int64
 UniProt              object
 Peptide              object
 PeptideAbundance    float64
 dtype: object,
 'Train Proteins': visit_id        object
 visit_month      int64
 patient_id       int64
 UniProt         object
 NPX            float64
 dtype: object,
 'Train Clinical Data': visit_id                                object
 patient_id                               int64
 visit_month                              int64
 updrs_1                                float64
 updrs_2                                float64
 updrs_3                                float64
 updrs_4                                float64
 upd23b_clinical_state_on_medication     object
 dtype: object,
 'Supplemental Clinical Data': visit_id                                object
 patient_id                               int64
 visit_month                              int64
 updrs_1                                f

1. Train Peptides:
    * visit_id, UniProt, Peptide: Object (String)
    * visit_month, patient_id: Integer
    * PeptideAbundance: Float
2. Train Proteins:
    * visit_id, UniProt: Object (String)
    * visit_month, patient_id: Integer
    * NPX: Float
3. Train Clinical Data:
    * visit_id, upd23b_clinical_state_on_medication: Object (String)
    * patient_id, visit_month: Integer
    * updrs_1 to updrs_4: Float
4. Supplemental Clinical Data:
    * visit_id, upd23b_clinical_state_on_medication: Object (String)
    * patient_id, visit_month: Integer
    * updrs_1 to updrs_4: Float

All the data types seem appropriate for their respective columns. String columns are represented as objects, numerical columns are either integers or floats based on the nature of the data they hold.

# Statistical overview

In [38]:
# Obtain basic statistics for numerical columns in train_peptides dataset
train_peptides_stats = train_peptides.describe()
train_peptides_stats

visit_month     patient_id  PeptideAbundance
count  981834.000000  981834.000000      9.818340e+05
mean       26.105061   32603.465361      6.428902e+05
std        22.913897   18605.934422      3.377989e+06
min         0.000000      55.000000      1.099850e+01
25%         6.000000   16566.000000      2.817425e+04
50%        24.000000   29313.000000      7.430830e+04
75%        48.000000   49995.000000      2.213388e+05
max       108.000000   65043.000000      1.787520e+08

In [39]:
# Obtain basic statistics for numerical columns in train_proteins dataset
train_proteins_stats = train_proteins.describe()
train_proteins_stats

visit_month     patient_id           NPX
count  232741.000000  232741.000000  2.327410e+05
mean       26.099205   32593.881873  2.712077e+06
std        22.874719   18608.479506  2.224155e+07
min         0.000000      55.000000  8.460820e+01
25%         6.000000   16566.000000  2.946440e+04
50%        24.000000   29313.000000  1.135560e+05
75%        48.000000   49995.000000  5.638940e+05
max       108.000000   65043.000000  6.138510e+08

The NPX values vary significantly, which might be expected given the diverse nature of protein expressions.


In [40]:
# Obtain basic statistics for numerical columns in train_clinical_data dataset
train_clinical_data_stats = train_clinical_data.describe()
train_clinical_data_stats

patient_id  visit_month      updrs_1     updrs_2      updrs_3   
count   2615.000000  2615.000000  2614.000000  2613.00000  2590.000000  \
mean   32651.743786    31.190822     7.110559     6.74359    19.421236   
std    18535.758700    25.199053     5.525955     6.32323    15.000289   
min       55.000000     0.000000     0.000000     0.00000     0.000000   
25%    16574.000000    10.500000     3.000000     1.00000     6.000000   
50%    29417.000000    24.000000     6.000000     5.00000    19.000000   
75%    50611.000000    48.000000    10.000000    10.00000    29.000000   
max    65043.000000   108.000000    33.000000    40.00000    86.000000   

           updrs_4  
count  1577.000000  
mean      1.861763  
std       3.022112  
min       0.000000  
25%       0.000000  
50%       0.000000  
75%       3.000000  
max      20.000000

The UPDRS scores range from 0 (no symptoms) to higher values indicating more severe symptoms. It's worth noting the variation in the scores, as this will be crucial when predicting disease progression.

Lastly, let's obtain the basic statistics for the supplemental_clinical_data dataset.

In [41]:
# Obtain basic statistics for numerical columns in supplemental_clinical_data dataset
supplemental_clinical_data_stats = supplemental_clinical_data.describe()
supplemental_clinical_data_stats

patient_id  visit_month      updrs_1      updrs_2      updrs_3   
count   2223.000000  2223.000000  2010.000000  2009.000000  2218.000000  \
mean   32478.016194    12.910481     5.684080     6.507715    22.917944   
std    18637.562796    13.060532     4.366964     4.968132    12.342596   
min       35.000000     0.000000     0.000000     0.000000     0.000000   
25%    16703.000000     0.000000     2.000000     2.000000    14.000000   
50%    32915.000000     6.000000     5.000000     5.000000    22.000000   
75%    47243.000000    24.000000     8.000000    10.000000    31.000000   
max    65530.000000    36.000000    27.000000    34.000000    72.000000   

           updrs_4  
count  1295.000000  
mean      0.840154  
std       1.860247  
min       0.000000  
25%       0.000000  
50%       0.000000  
75%       0.000000  
max      12.000000

Similar to the train_clinical_data, the UPDRS scores in the supplemental dataset also range from 0 (no symptoms) to higher values indicating more severe symptoms.